### ISSUE TITLE
unable to use the push plugin with android@6.3.0. receive error "Could not find method google() for arguments [] on repository container."

### ISSUE THREAD

**Issue by mkhalife** [Microsoft/appcenter-sdk-cordova#55]
### **Description**

receive build error below when using the push plugin with android@6.3.0 (which is the default for phonegap cli 7.1.0)

> FAILURE: Build failed with an exception.
> 
> * Where:
> Script '/cordova-plugin-appcenter-push/workplace-AppCenterPush.gradle' line: 3
> 
> * What went wrong:
> A problem occurred evaluating script.
> > Could not find method google() for arguments [] on repository container.
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
>     at ChildProcess.whenDone (/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
>     at emitTwo (events.js:106:13)
>     at ChildProcess.emit (events.js:191:7)
>     at maybeClose (internal/child_process.js:877:16)
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)

I simply upgraded to 6.4 so it's not an issue for me anymore but thought i'd report it.

### **Repro Steps**

1. install phonegap with `npm install -g phonegap`
2. `phonegap create test`
3. `cd test`
4. `phonegap plugin add cordova-plugin-appcenter-push`
5. `phonegap platform add android@6.3.0`
6. add your google-services.json file to the project root
7. add this line to your config.xml
`<resource-file src="google-services.json" target="app/google-services.json" />`
8. remove this line from your config.xml (you'll receive an error otherwise)
`<preference name="android-minSdkVersion" value="14" />`
9. `phonegap build android --verbose`

### **Details**

1. Which SDK version are you using?
    - latest
2. Which OS version did you experience the issue on?
    - N/A
3. What version of Cordova, Cordova platforms, Cocoapods, and Ruby are you using?
    - Cordova version 7.1.0
4. What device version did you see this error on?  Were you using a simulator or a physical device?
    - N/A
5. What third party plugins are you using?
    - N/A

**Comment by annakocheshkova** [comment]
Hi @mkhalife! Happy that the issue got resolved, but if for some reason you still want to use android `6.3`, here's what I did:

1. Complete the repro steps you priovided and got the error
2. Updated gradle version in `platforms/android/build.gradle`:
```
dependencies {
     classpath 'com.android.tools.build:gradle:2.3.3'
}
...
task wrapper(type: Wrapper) {
    gradleVersion = '4.6'
}
```
3. Run `$Env:CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL="https\://services.gradle.org/distributions/gradle-4.6-all.zip"`
4. Build

Why so? 
Because the `google()` repo (which is a shortcut to Google's Maven repository for dependencies, which is used in `cordova-plugin-appcenter-push`), was introduced with gradle v.4.0. It also requires gradle plugin for Android 3.x.
This is why, when you upgrade to android `6.4` (where gradle is upgraded as well), it builds.
